airbyte-integrations/connectors/source-zendesk-chat/source_zendesk_chat/spec.json

```diff
@@ -4,20 +4,82 @@
     "$schema": "http://json-schema.org/draft-07/schema#",
     "title": "Zendesk Chat Spec",
     "type": "object",
-    "required": ["start_date", "access_token"],
-    "additionalProperties": false,
+    "required": ["start_date", "credentials"],
+    "additionalProperties": true,
     "properties": {
       "start_date": {
         "type": "string",
+        "title": "Start Date",
         "description": "The date from which you'd like to replicate data for Zendesk Chat API, in the format YYYY-MM-DDT00:00:00Z.",
         "examples": ["2021-02-01T00:00:00Z"],
         "pattern": "^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}Z$"
       },
-      "access_token": {
-        "type": "string",
-        "description": "The value of the Access Token generated. See the <a href=\"https://docs.airbyte.io/integrations/sources/zendesk-chat\">docs</a> for more information",
-        "airbyte_secret": true
+      "credentials": {
+        "title": "Authorization Method",
+        "type": "object",
+        "oneOf": [
+          {
+            "type": "object",
+            "title": "OAuth2.0",
+            "required": ["credentials"],
+            "properties": {
+              "credentials": {
+                "type": "string",
+                "const": "oauth2.0",
+                "enum": ["oauth2.0"],
+                "default": "oauth2.0",
+                "order": 0
+              },
+              "client_id": {
+                "type": "string",
+                "title": "Client ID",
+                "description": "The Client ID of your OAuth application",
+                "airbyte_secret": true
+              },
+              "client_secret": {
+                "type": "string",
+                "title": "Client Secret",
+                "description": "The Client Secret of your OAuth application.",
+                "airbyte_secret": true
+              },
+              "access_token": {
+                "type": "string",
+                "title": "Access Token",
+                "description": "Access Token for making authenticated requests.",
+                "airbyte_secret": true
+              }
+            }
+          },
+          {
+            "type": "object",
+            "title": "Access Token",
+            "required": ["credentials"],
+            "properties": {
+              "credentials": {
```

Comment:
I would rename this property to authe_method or something like that,

Otherwise json path is ../credentials[0]/credentials

Reply:
Thats totally ok